pkg/endpointconn/endpoint.go

```diff
@@ -18,7 +18,7 @@ func Get(ctx context.Context, endpoint string) (*grpc.ClientConn, error) {
 		return nil, err
 	}
 
-	conn, err := grpc.DialContext(ctx, target, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithContextDialer(dialer))
+	conn, err := grpc.DialContext(ctx, target, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithContextDialer(dialer), grpc.FailOnNonTempDialError(true))
```

Comment:
`grpc.FailOnNonTempDialError` hints gRPC to stop retrying on things that aren't going to fix themselves